sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -2549,6 +2551,47 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("SPARK-40045: Move the post-Scan Filters to the far right") {
+    val t1 = s"${catalogAndNamespace}table"
+    withTable(t1) {
+      spark.udf.register("udfStrLen", (str: String) => str.length)
```

Comment:
nit: wrap the test with `withUserDefinedFunction` to unregister the function at the end.

Reply:
> nit: wrap the test with `withUserDefinedFunction` to unregister the function at the end.

@cloud-fan Thanks. Updated